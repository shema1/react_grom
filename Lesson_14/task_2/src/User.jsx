import React from 'react';
import { getUserList } from './userGateway.js';

class User extends React.Component {
    state= {
        userData: null
    };

    componentDidMount() {
        getUserList().then(userData => 
            this.setState({
                userData,
            }),
        );
    };

    render() {
        const { userData } = this.state;
        if (!userData) {
            return null;
        }
        const { name, location, avatar_url } = userData;
        return (
            <div className="user">
                <img
                    alt="User Avatar"
                    src={avatar_url}
                    className="user__avatar"
                />
                <div className="user__info">
                    <span className="user__name">{name}</span>
                    <span className="user__location">{location}</span>
                </div>
            </div>
        );
    }
}

export default User;