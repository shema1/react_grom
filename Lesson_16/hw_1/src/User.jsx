import React from "react";

const User = ({ match }) => {
  return (
    <div className="user">
      <img
        alt="User Avatar"
        src={
          match.params.userId === "facebook"
            ? "https://lh3.googleusercontent.com/ccWDU4A7fX1R24v-vvT480ySh26AYp97g1VrIB_FIdjRcuQB2JP2WdY7h_wVVAeSpg"
            : "https://avatars1.githubusercontent.com/u/9919?v=4"
        }
        className="user__avatar"
      />
      <div className="user__info">
        <span className="user__name">{match.params.userId}</span>
        <span className="user__location">
          {" "}
          {match.params.userId === "facebook"
            ? "Menlo Park, CA"
            : "San Francisco,CA"}
        </span>
      </div>
    </div>
  );
  ("https://avatars1.githubusercontent.com/u/9919?v=4");
};

export default User;
