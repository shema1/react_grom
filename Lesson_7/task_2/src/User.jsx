import React from "react";
import UserList from "./UserList";

const User = ({ name, age }) => {
  return (
    <li key={age} className="user">
      <span className="user__name">{name}</span>
      <span className="user__age">{age}</span>
    </li>
  );
};

export default User;
