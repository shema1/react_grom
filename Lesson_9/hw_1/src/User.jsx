import React, { Component } from "react";

class User extends Component {
  render() {
    const { name, age } = this.props;
    return (
      <li className="user">
        <span className="user__name">{name}</span>
        <span className="user__age">{age}</span>
      </li>
    );
  }
}

export default User;
