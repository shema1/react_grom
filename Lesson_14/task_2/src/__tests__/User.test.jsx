import React from 'react';
import { shallow } from 'enzyme';
import User from '../User.jsx';
import { getUserList }  from '../userGateway.js';
jest.mock('../userGateway', () => {
    return {
        getUserList: jest.fn(() => Promise.resolve())
    }
})

describe('User', () => {
    it('should display Login button if user logged out', () => {
        const wrappedComponent = shallow(<User/>);
        expect(wrappedComponent.find('.user').exists()).toEqual(false);
    });

    it('should display Login button if user logged out', () => {
        const wrappedComponent = shallow(<User/>);
        wrappedComponent.setState({
            userData: {
                avatar__url: 'https://example.com',
                name:'John',
                location:'England'
            }
        })
        expect(wrappedComponent.find('.user')).toMatchSnapshot(true);
    });

    it('should display Login button if user logged out', () => {
        const wrappedComponent = shallow(<User/>);
        expect(getUserList).toBeCalled();
    });
})