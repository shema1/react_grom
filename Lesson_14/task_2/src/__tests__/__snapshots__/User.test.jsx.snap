// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should display Login button if user logged out 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      John
    </span>
    <span
      className="user__location"
    >
      England
    </span>
  </div>
</div>
`;
